snuba/utils/schemas.py

```diff
@@ -702,3 +705,41 @@ def set_modifiers(self, modifiers: Optional[TModifiers]) -> Enum[TModifiers]:
 
     def get_raw(self) -> Enum[TModifiers]:
         return Enum(self.values)
+
+
+class ColumnValidator:
+    def __init__(self, column_set: ColumnSet):
+        self._column_set = column_set
+
+    def validate(self, column_name: str, values: Sequence[AnyType]) -> None:
+        expected_type = self._column_set[column_name].type
+        val_func: Optional[Callable[[AnyType], None]]
+        match expected_type:
+            case UUID():
+                val_func = self._validate_uuid
+            case Int():
+                val_func = self._validate_int
+            case UInt():
+                val_func = self._validate_int
```

Comment:
I am trying to understand the support of unsigned integers in the `schemas.py` - https://github.com/getsentry/snuba/blob/d4b164c06710e15bed33d868102305adf8fe495c/snuba/utils/schemas.py#L539 but it looks virtually similar to integer. IMO, validating unsigned integers in the same way as integers is most probably wrong. One way I can think of is to use `ctypes` - https://docs.python.org/3/library/ctypes.html#ctypes.c_uint but curious to know what you think.